pkg/v1/tkg/web/src/app/views/landing/wizard/shared/components/steps/metadata-step/metadata-step.component.ts

```diff
@@ -36,48 +49,104 @@ export class MetadataStepComponent extends StepFormDirective implements OnInit {
             const savedLabelsArray = savedLabelsString.split(', ')
             savedLabelsArray.map(label => {
                 const labelArray = label.split(':');
-                this.labels.set(labelArray[0], labelArray[1]);
+                this.addLabel(labelArray[0], labelArray[1]);
+                this.savedKeySet.add(LABEL_KEY_NAME + this.labelCounter);
             });
         }
         super.initFormWithSavedData();
     }
 
-    addLabel(key: string, value: string) {
-        if (key === '' || value === '') {
-            this.errorNotification = `Key and value for Labels are required.`;
-        } else if (!this.labels.has(key)) {
-            this.labels.set(key, value);
-            this.formGroup.get('clusterLabels').setValue(this.labels);
-            this.formGroup.controls['newLabelKey'].setValue('');
-            this.formGroup.controls['newLabelValue'].setValue('');
-        } else {
-            this.errorNotification = `A Label with the same key already exists.`;
-        }
+    addLabel(key?: string, value?: string) {
+        this.labelCounter++;
+        this.labels.set(LABEL_KEY_NAME + this.labelCounter, LABEL_VALUE_NAME + this.labelCounter);
+        this.keySet.add(LABEL_KEY_NAME + this.labelCounter);
+        FormUtils.addControl(
+            this.formGroup,
+            LABEL_KEY_NAME + this.labelCounter,
+            new FormControl(key || '', [
+                this.validationService.isValidLabelOrAnnotation(),
+                this.validationService.isUniqueLabel(
+                    this.formGroup,
+                    this.keySet,
+                    LABEL_KEY_NAME + this.labelCounter)
+            ])
+        );
+
+        FormUtils.addControl(
+            this.formGroup,
+            LABEL_VALUE_NAME + this.labelCounter,
+            new FormControl(value || '', [
+                this.validationService.isValidLabelOrAnnotation()
+            ])
+        );
+        // Label value depends on Label key. e.g.: if label key is not empty, then label value is required
+        this.onChangeWithDependentField(LABEL_KEY_NAME + this.labelCounter, LABEL_VALUE_NAME + this.labelCounter);
+        // Label key depends on Label value. e.g.: if label value is not empty, then label key is required
+        this.onChangeWithDependentField(LABEL_VALUE_NAME + this.labelCounter, LABEL_KEY_NAME + this.labelCounter);
+        this.validateAllLabels();
+    }
+    /**
```

Comment:
Please fix line spacing here. Add a line break between 87 and 88, then remove line break on 94

Reply:
added